arcgis-ios-sdk-samples/Utility network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -53,31 +49,22 @@ class PerformValveIsolationTraceViewController: UIViewController {
     var traceConfiguration: AGSUtilityTraceConfiguration?
     var selectedCategory: AGSUtilityCategory?
     var startingLocationElement: AGSUtilityElement!
-    var utilityNetwork: AGSUtilityNetwork!
-    
-    /// The gas line layer ./3 and gas device layer ./0 are created from the feature service URL.
-    lazy var layers: [AGSFeatureLayer] = {
-        let urls = [
-            featureServiceURL.appendingPathComponent("3"),
-            featureServiceURL.appendingPathComponent("0")
-        ]
-        return urls.map { (url) -> AGSFeatureLayer in
-            let featureTable = AGSServiceFeatureTable(url: url)
-            return AGSFeatureLayer(featureTable: featureTable)
-        }
-    }()
+    let utilityNetwork = AGSUtilityNetwork(url: featureServiceURL)
+    let serviceGeodatabase = AGSServiceGeodatabase(url: featureServiceURL)
+    var layers = [AGSFeatureLayer]()
     
     // MARK: Initialize map and utility network
     
     /// Create a map.
     ///
     /// - Parameter layers: The feature layers for the utility network.
-    /// - Returns: An `AGSMap` object.
-    func makeMap(with layers: [AGSFeatureLayer]) -> AGSMap {
+    func setMap(with layers: [AGSFeatureLayer]) {
```

Comment:
Now that creation of the utility network doesn't depend on the map, could we go back to using the `makeMap` method?

Reply:
And since both the `utilityNetwork` and `layers` are properties, probably we can remove the parameters from this method.

Reply:
Hm, I could use the `makeMap` method but would have to remove the layers parameter since the service geodatabase must be loaded before I can instantiate the layers array. Please let me know what you think!

Reply:
Yeah, I guess it makes sense to remove the layers parameter. Since the layers need to be added later, it makes sense to do that through the `map` property of the map view, just like you did.